packages/amplify-migration-e2e/src/__tests__/migration_codegen_e2e.test.ts

```diff
@@ -0,0 +1,110 @@
+import path from 'node:path';
+import assert from 'node:assert';
+import { createNewProjectDir, npmInstall } from '@aws-amplify/amplify-e2e-core';
+import { createGen2Renderer } from '@aws-amplify/amplify-gen2-codegen';
+import { copyFunctionFile } from '../function_utils';
+import { copyGen1Schema } from '../api_utils';
+import {
+  cleanupProjects,
+  setupAndPushDefaultGen1Project,
+  setupAndPushAuthWithMaxOptionsGen1Project,
+  setupAndPushStorageWithMaxOptionsGen1Project,
+  runCodegenCommand,
+  runGen2SandboxCommand,
+} from '..';
+import {
+  assertStorageWithMaxOptionsGen1Setup,
+  assertAuthWithMaxOptionsGen1Setup,
+  assertDefaultGen1Setup,
+  assertUserPoolResource,
+  assertStorageResource,
+  assertFunctionResource,
+  assertDataResource,
+  assertIdentityPoolResource,
+  assertUserPoolClientsResource,
+} from '../assertions';
+import { removeErrorThrowsFromAuthResourceFile } from '../auth_utils';
+import { updatePackageDependency } from '../updatePackageJson';
+import { toggleSandboxSecrets } from '../secrets';
+
+void describe('Codegen E2E tests', () => {
+  void describe('render pipeline', () => {
+    void it('renders a project with no parameters', async () => {
+      const pipeline = createGen2Renderer({
+        outputDir: path.join(process.env.INIT_CWD ?? './', 'output'),
+        auth: {
+          loginOptions: {
+            email: true,
+          },
+        },
+      });
+      await assert.doesNotReject(pipeline.render);
+    });
+  });
+  void describe('Full Migration Codegen Flow', () => {
+    let projRoot: string;
+    let projName: string;
+
+    beforeEach(async () => {
+      const baseDir = process.env.INIT_CWD ?? process.cwd();
+      projRoot = await createNewProjectDir('codegen_e2e_flow_test', path.join(baseDir, '..', '..'));
+      projName = `test${Math.floor(Math.random() * 1000000)}`;
+    });
+
+    afterEach(async () => {
+      await cleanupProjects(projRoot);
+    });
+
+    void it('should init a project & add auth, function, storage, api with defaults & perform full migration codegen flow', async () => {
+      await setupAndPushDefaultGen1Project(projRoot, projName);
+      const { gen1UserPoolId, gen1ClientIds, gen1IdentityPoolId, gen1FunctionName, gen1BucketName, gen1GraphqlApiId, gen1Region } =
+        await assertDefaultGen1Setup(projRoot);
+      await runCodegenCommand(projRoot);
+      await copyFunctionFile(projRoot, 'function', gen1FunctionName);
+      await copyGen1Schema(projRoot, projName);
+      await updatePackageDependency(projRoot, '@aws-amplify/backend', '0.0.0-test-20241003180022');
```

Comment:
why not latest version?

Reply:
The backend library included dataTableMapping in the data props only in the tagged PR release. I'll check in with the team to know if it's published.

Reply:
Checked in with the team, they are still working on a new release.

Reply:
ah I see, pls add a TODO and file a bug to ensure we replace it with latest when they are ready

Reply:
Okay, will do.